Add eshell-bookmark

Fuco1 opened the pull request:
### Brief summary of what the package does

Allows you to store bookmarks to eshell buffers

### Direct link to the package repository

https://github.com/Fuco1/eshell-bookmark

### Your association with the package

maintainer

### Checklist

Please confirm with `x`:

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses). 
- [x] I've read [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)
- [x] I've used [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [ ] I've built and installed the package using the instructions in [CONTRIBUTING.md](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.md)

Fuco1 commented:
Why isn't the lint check in the CI build?

purcell commented:
> Why isn't the lint check in the CI build?

It's a bit fiddly to integrate, so for now it's just in the PR checklist. Additionally, `package-lint` doesn't cover multi-file packages, so the build output would just be for information purposes: I'm not sure it could easily be used to fail the build.

manuel-uberti commented:
Any chance for this one to be merged?

Fuco1 commented:
@manuel-uberti I still haven't got to addressing the issues @purcell outlined above.  I'll try to do it over the weekend... I'm kind of bad with focus lately :/

Fuco1 commented:
Okay I've just done it now because I would forget.

manuel-uberti commented:
Thank you!

purcell commented:
Super, thanks!
